meson.build

```diff
@@ -534,10 +534,24 @@ if thread_local_storage
   tls_model_spec = '%{!ftls-model:-ftls-model=' + get_option('tls-model') + '}'
 endif
 
+# specs_prefix_format must point to the toolchain's root directory
 if sysroot_install
   sysroot = run_command(cc.cmd_array() + ['-print-sysroot'], check : true).stdout().split('\n')[0]
   if sysroot != ''
-    specs_prefix_format = '%R/@0@'
+    sysroot_realpath = run_command('realpath', sysroot, check : true).stdout().split('\n')[0]
```

Comment:
This needs to work even if realpath isn't available. Is there some way you could use fs.is_samepath and make this work with only internal meson functions?

Reply:
@keith-packard, sorry, but I don't see any options other than writing ugly spaghetti code to resolve this using only "meson" built-in calls. Feel free to close the PR. I might come back with a PR once Meson has improved (I hope).

Reply:
@keith-packard , please check updated commit